### PR TITLE
fix(fsevents): invalid gc usage

### PR DESCRIPTION
**Pull request by rgrinberg**
make sure runloop pointer is always registered

### CODE REVIEW

File: src/fsevents/fsevents_stubs.c [hunk 1]
```diff
@@ -210,21 +210,17 @@ CAMLprim value dune_fsevents_stop(value v_t) {
...
   CAMLparam1(v_t);
   dune_fsevents_t *t = (dune_fsevents_t *)Nativeint_val(v_t);
   if (t->runloop == NULL) {
     CAMLreturn(Val_int(0));
   } else {
-    CAMLreturn(Val_some(caml_copy_nativeint((intnat)t->runloop)));
+    CAMLlocal2(v_some, v_runloop);
+    v_runloop = caml_copy_nativeint((intnat)t->runloop);
+    v_some = caml_alloc_small(1, 0);
+    Field(v_some, 0) = v_runloop;
```
Comment (nojb) [inline src/fsevents/fsevents_stubs.c:222]: Shouldn't this be `Store_field(v_some, 0, v_runloop)`?

**Review by nojb — COMMENTED** [review]
I had missed that `Val_some` was defined in your code and that it was registering the argument as a root. So I don't think there is anything wrong with the use of `Val_some` in the existing code after all. But I think one should use `Store_field` instead of using `Field(v_some, 0)` directly as an l-value.

File: src/fsevents/fsevents_stubs.c (hunk 1 above)
Comment (rgrinberg) [reply src/fsevents/fsevents_stubs.c]: Fixed.

**Comment by rgrinberg** [comment]
Indeed. I fixed the message to say that I'm not fixing anything but refactoring. The new code is still cleaner because `Val_some` was only used once.

File: src/fsevents/fsevents_stubs.c [hunk 2]
```diff
@@ -209,21 +209,17 @@ CAMLprim value dune_fsevents_stop(value v_t) {
...
-}
-
 CAMLprim value dune_fsevents_runloop_get(value v_t) {
   CAMLparam1(v_t);
   dune_fsevents_t *t = (dune_fsevents_t *)Nativeint_val(v_t);
   if (t->runloop == NULL) {
     CAMLreturn(Val_int(0));
   } else {
-    CAMLreturn(Val_some(caml_copy_nativeint((intnat)t->runloop)));
+    CAMLlocal2(v_some, v_runloop);
```
Comment (nojb) [inline src/fsevents/fsevents_stubs.c:218]: Having the `CAMLlocal2` in a block is somewhat unorthodox, but I guess it works?

Comment (rgrinberg) [reply src/fsevents/fsevents_stubs.c:218]: If you think it's suspicious I can move it to the top.

Comment (nojb) [reply src/fsevents/fsevents_stubs.c:218]: Sure, let's move it to the top! Thanks.

**Review by nojb — APPROVED** [review]
LGTM (modulo the `CAMLlocal` issue).

**Pull request merged**